packages/types/src/sszTypes.ts

```diff
@@ -1,13 +1,192 @@
+
+const executionForks: ForkExecution[] = [ForkName.bellatrix, ForkName.capella, ForkName.deneb];
+const lightCLientForks: ForkLightClient[] = [ForkName.altair, ForkName.bellatrix, ForkName.capella, ForkName.deneb];
+const blobsForks: ForkBlobs[] = [ForkName.deneb];
+
+export const allForksExecution = pick(typesByFork, ...executionForks);
+export const allForksLightClient = pick(typesByFork, ...lightCLientForks);
+export const allForksBlobs = pick(typesByFork, ...blobsForks);
+export const allForksBlinded = {
+  bellatrix: {
+    BeaconBlockBody: bellatrix.BlindedBeaconBlockBody,
+    BeaconBlock: bellatrix.BlindedBeaconBlock,
+    SignedBeaconBlock: bellatrix.SignedBlindedBeaconBlock,
+  },
+  capella: {
+    BeaconBlockBody: capella.BlindedBeaconBlockBody,
+    BeaconBlock: capella.BlindedBeaconBlock,
+    SignedBeaconBlock: capella.SignedBlindedBeaconBlock,
+  },
+  deneb: {
+    BeaconBlockBody: deneb.BlindedBeaconBlockBody,
+    BeaconBlock: deneb.BlindedBeaconBlock,
+    SignedBeaconBlock: deneb.SignedBlindedBeaconBlock,
+  },
+};
+
+// TODO: These helpers should be removed along with `allForksBlinded`
+type SSZBlindedTypesByFork = {
+  [F in keyof typeof allForksBlinded]: {
+    [T in keyof (typeof allForksBlinded)[F]]: (typeof allForksBlinded)[F][T];
+  };
+};
+
+// TODO: These helpers should be removed along with `allForksBlinded`
+export type SSZBlindedTypesFor<
+  F extends ForkExecution,
+  K extends keyof SSZBlindedTypesByFork[F] | void = void,
+> = K extends void
+  ? // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    {[K2 in keyof SSZBlindedTypesByFork[F]]: UnionForksTypeOf<SSZBlindedTypesByFork[F][K2]>}
+  : // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    UnionForksTypeOf<SSZBlindedTypesByFork[F][Exclude<K, void>]>;
+
+/**
+ * A type of union of forks must accept as any parameter the UNION of all fork types.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type UnionForksTypeOf<UnionOfForkTypes extends ContainerType<any>> = CompositeType<
+  ValueOf<UnionOfForkTypes>,
+  CompositeView<UnionOfForkTypes>,
+  CompositeViewDU<UnionOfForkTypes>
+>;
+
+type SSZTypesByFork = {
+  [F in keyof typeof typesByFork]: {
+    [T in keyof (typeof typesByFork)[F]]: (typeof typesByFork)[F][T];
+  };
+};
+
+export type SSZTypesFor<F extends ForkName, K extends keyof SSZTypesByFork[F] | void = void> = K extends void
+  ? // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    {[K2 in keyof SSZTypesByFork[F]]: UnionForksTypeOf<SSZTypesByFork[F][K2]>}
+  : // It compiles fine, need to debug the error
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
```

Comment:
When we touch the limitations of TS, such cases occurs and in past as well and resolved with findings some hack or with TS upgrade. In this case explicitly, it's a utility type and it behaves as expected even with that compile error. So why we not considering this error a blocker,

I see no difference of having `@ts-expect-error` on runtime code or type code. The only thing matter is one understand clearly the error and make sure is not impacting the downstream code.

Reply:
Just about the comment "need to debug the error", since are not debugging the error might wanna add a more useful comment

Reply:
> Just about the comment "need to debug the error", since are not debugging the error might wanna add a more useful comment

I will definitely debug, discuss on TS forums and may be also report to TS.